src/NexusMods.App.UI/Pages/Downloads/InProgressViewModel.cs

```diff
@@ -127,37 +142,103 @@ public class InProgressViewModel : APageViewModel<IInProgressViewModel>, IInProg
-            .ToObservableChangeSet(x => x.PersistentState.Id)
+        var tasksChangeSet = _downloadService.Downloads
+            .ToObservableChangeSet(x => x.PersistentState.Id);
+        
+        InProgressTaskChangeSet = tasksChangeSet
             .Transform(x =>
                 {
                     var vm = new DownloadTaskViewModel(x);
                     vm.Activator.Activate();
                     return (IDownloadTaskViewModel)vm;
                 }
             )
-            .FilterOnObservable((item, key) => item.WhenAnyValue(v => v.Status)
-                .Select(s => s != DownloadTaskStatus.Cancelled && s != DownloadTaskStatus.Completed))
+            .FilterOnObservable((item, key) =>
+                {
+                    return item.WhenAnyValue(v => v.Status)
+                        .Select(s => s != DownloadTaskStatus.Cancelled && s != DownloadTaskStatus.Completed);
+                }
+            )
+            .AutoRefreshOnObservable(task =>
+                {
+                    return task.WhenAnyValue(x => x.Status);
+                }
+            )
             .DisposeMany()
             .OnUI();
+            
+
+        CompletedTaskChangeSet = tasksChangeSet
+            .Transform(x =>
+                {
+                    var vm = new DownloadTaskViewModel(x);
+                    vm.HideCommand = ReactiveCommand.CreateFromTask(async () => await HideTasks(true, [vm]));
+                    vm.ViewInLibraryCommand = ReactiveCommand.Create<NavigationInformation>((navInfo) =>
+                    {
+                        var controller = GetWorkspaceController();
+                        var workspaces = controller
+                            .AllWorkspaces
+                            .Where(w => w.Context is LoadoutContext loadoutContext
+                                        && conn.Db.Get<Loadout.Model>(loadoutContext.LoadoutId.Value).Installation.Game.Domain.Equals(vm.Game)
+                            )
+                            .Select(w => (w.Id, Context: (LoadoutContext)w.Context)).ToArray();
+                        
+                        if (workspaces.Length == 0)
+                            return;
+                        
+                        var workspace = workspaces[0];
+
+                        var pageData = new PageData
+                        {
+                            FactoryId = FileOriginsPageFactory.StaticId,
+                            Context = new FileOriginsPageContext { LoadoutId = workspace.Context.LoadoutId },
+                        };
+                        var behavior = GetWorkspaceController().GetOpenPageBehavior(pageData, navInfo, Optional<PageIdBundle>.None);
+                        
+                        controller.OpenPage(workspace.Id, pageData, behavior);
+                        controller.ChangeActiveWorkspace(workspace.Id);
+                    });
+                    vm.Activator.Activate();
```

Comment:
This is a very bold statement, why is this here?

Reply:
Do you mean the Activator thing? That was there before. From my understanding `IDownloadTaskViewModel` is a fake VM that doesn't actually have a view, the individual columns take data from it, but not as the VM. Their VM is either `InProgressViewModel` or the column definition VM. 
Calling Activate there triggers the `WhenActivated()` of the VM. Personally I